app/controllers/users.js

```
@@ -0,0 +1,38 @@
+const users = require('../models').users,
+  bcrypt = require('bcryptjs'),
+  helpers = require('../helpers'),
+  logger = require('../logger'),
+  errors = require('../errors');
+
+exports.signUp = (req, res, next) => {
+  const saltRounds = 5;
+  const signErrors = [];
+
+  if (!helpers.validatePassword(req.body.password)) signErrors.push(errors.invalidPassword());
+  if (!helpers.validateEmail(req.body.email)) signErrors.push(errors.invalidEmail());
+
+  if (signErrors.length) {
+    throw signErrors;
```

and here you can throw the generic error or call next with the error